tests/test_devices.py

```diff
@@ -255,6 +255,152 @@ def test_device_inventory_update_success(client, mocker):
     assert device._inventory == inventory
 
 
+class test_inputs:
+    def test_property_populated(client, mocker):
+        """Should check if the device property are correctly populated"""
+        device = AlarmDevice(client)
+        mocker.spy(device._connection, "query")
```

Comment:
I'm not sure you need to spy "query". You are correctly not testing the `_connection` object as this test is for the property only, so I guess you can remove it. If you want to check a bit `pytest` documentation to see when you should use Mock and Spy and what is the difference between the two here's the link that can help you! https://pytest-mock.readthedocs.io/en/latest/usage.html

My suggestion is to remove this line as it does not change the test result.